pkg/services/live/pipeline/frame_storage.go

```diff
@@ -0,0 +1,37 @@
+package pipeline
+
+import (
+	"sync"
+
+	"github.com/grafana/grafana/pkg/services/live/orgchannel"
+
+	"github.com/grafana/grafana-plugin-sdk-go/data"
+)
+
+// FrameStorage keeps last channel frame in memory. Not usable in HA setup.
+type FrameStorage struct {
+	mu     sync.RWMutex
+	frames map[string]*data.Frame
+}
+
+func NewFrameStorage() *FrameStorage {
+	return &FrameStorage{
+		frames: map[string]*data.Frame{},
+	}
+}
+
+func (s *FrameStorage) Set(orgID int64, channel string, frame *data.Frame) error {
```

Comment:
Yeah, we still need to think more on HA case for this frame storage. So maybe this will be kept in Redis eventually, or we will be able to reuse our existing schema cache in managed stream and keep frame there. Storage is one of the most difficult things here so we will address it somehow.